x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlExecutionInfo.java

```diff
@@ -67,17 +66,20 @@ public class EsqlExecutionInfo implements ChunkedToXContentObject, Writeable {
     // Updates to the Cluster occur with the updateCluster method that given the key to map transforms an
     // old Cluster Object to a new Cluster Object with the remapping function.
     public final Map<String, Cluster> clusterInfo;
-    private TimeValue overallTook;
     // whether the user has asked for CCS metadata to be in the JSON response (the overall took will always be present)
     private final boolean includeCCSMetadata;
 
     // fields that are not Writeable since they are only needed on the primary CCS coordinator
     private final transient Predicate<String> skipUnavailablePredicate;
-    private final transient Long relativeStartNanos;  // start time for an ESQL query for calculating took times
-    private transient TimeValue planningTookTime;  // time elapsed since start of query to calling ComputeService.execute
     private volatile boolean isPartial; // Does this request have partial results?
     private transient volatile boolean isStopped; // Have we received stop command?
 
+    // start time for the ESQL query for calculating time spans relative to the beginning of the query
+    private final transient TimeSpan.Builder relativeStart;
+    private transient TimeSpan overallTimeSpan;
+    private transient TimeSpan planningTimeSpan; // time elapsed since start of query to calling ComputeService.execute
```

Comment:
Okay, this actually need to be serialized in order to pass this information when executing query async